datalake/src/main/java/com/sequenceiq/datalake/service/recovery/RecoveryService.java

```diff
@@ -1,47 +1,45 @@
 package com.sequenceiq.datalake.service.recovery;
 
-import javax.inject.Inject;
-
-import org.springframework.stereotype.Service;
-
 import com.sequenceiq.cloudbreak.api.endpoint.v4.dto.NameOrCrn;
-import com.sequenceiq.datalake.service.resize.recovery.ResizeRecoveryService;
-import com.sequenceiq.datalake.service.upgrade.recovery.SdxUpgradeRecoveryService;
 import com.sequenceiq.sdx.api.model.SdxRecoverableResponse;
-import com.sequenceiq.sdx.api.model.SdxRecoveryRequest;
+import com.sequenceiq.sdx.api.model.UpgradeRecoveryRequest;
 import com.sequenceiq.sdx.api.model.SdxRecoveryResponse;
 
-@Service
 /**
- * Recovers an SDX cluster from a failure during Resize or Upgrade.
+ * Defines a service that can be used to recover CB deployed infrastructure.
+ *
+ * Services that can recovery failed infrastructure must be able to <em>validate</em> that a recoveyr action is appropriate and
+ * be able to <em>trigger</em> the recovery.
  *
- * Chooses the appropriate recovery action to take based on the state of the SDX cluster.
- * <ul>
- *     <li>{@Code SdxUpgradeRecoveryService}</li>
- *     <li>{@Code ResizeRecoveryService}</li>
- * </ul>
+ * The Recovery Service is parameterized by the type of recovery request made to it. The recovery request should be a value object representing
+ * an HTTP response object.
+ * @param <T> entity representing additional properties for the recovery
  */
-public class RecoveryService {
-    @Inject
-    private SdxUpgradeRecoveryService sdxUpgradeRecoveryService;
-
-    @Inject
-    private ResizeRecoveryService resizeRecoveryService;
-
-    // todo: refactor this to return a flow identifier
-    public SdxRecoveryResponse triggerRecovery(String crn, NameOrCrn nameOrCrn, SdxRecoveryRequest sdxRecoveryRequest) {
-        if (resizeRecoveryService.canRecover()) {
-            return resizeRecoveryService.triggerRecovery();
-        } else {
-            return sdxUpgradeRecoveryService.triggerRecovery(crn, nameOrCrn, sdxRecoveryRequest);
-        }
-    }
+public interface RecoveryService<T extends UpgradeRecoveryRequest> {
```

Comment:
With the renames going on here, I'm not sure if UpgradeRecoveryRequest is going to have something specific to upgrades and there will be a ResizeRecoveryRequest as well. Or if one request type will work for both.